### PR TITLE
Customizable start chat error pane

### PR DESCRIPTION
**Pull request by kevinlin98z**
### Description
Enable full set of customizations for the start chat error pane.

## Solution Proposed
- Leverage the existing `LoadingPane` component as the underlying component for the new `StartChatErrorPane` component (`StartChatErrorPaneStateful`)
- Remove any ties to error pane from `LoadingPaneStateful`
- `StartChatErrorPaneProps` contains the customizations to be applied generically across all error string panes, and the different localization/storage of specific error strings 
- Default error pane props are applied in stateful
- Dispatching and managing state based on `ConversationState.Error` to determine showing the `StartChatErrorPane`

### Acceptance criteria
The elements of the start chat error pane such as title, subtitle, icon, and container associated styles can be customized. Existing conversation state flows are not broken.

## Test cases and evidence
Default:
image

Customized:
image
image
image

Demo video: https://microsoft-my.sharepoint.com/:v:/p/kevinlin/EQhiLUb8p-5EpdXWc1LoRnQB7GTlPMMNZk3vdaj5s_-ysA?e=TcmWpn

Tested on few android / iOS devices & browsers, see below for some samples: https://microsoft-my.sharepoint.com/:f:/p/kevinlin/Eq6xXZf4pElGqWnyzt783BEBFwnRFOohp881DwptHnf9Lw?e=tcW8xG

### Sanity Tests
-   [x] You have tested all changes in Popout mode
-   [x] You have tested all changes in cross browsers i.e Edge, Chrome, Firefox, Safari and mobile devices(iOS and Android)
-   [x] Your changes are included in the [CHANGELOG](../CHANGE_LOG.md)

## A11y
- [x] You have run the [Automated Accessibility Check](https://accessibilityinsights.io/docs/en/windows/getstarted/automatedchecks) and have no reported issues

__*Please provide justification if any of the validations has been skipped.*__

### CODE REVIEW

File: chat-widget/src/components/livechatwidget/interfaces/ILiveChatWidgetProps.ts [hunk 1]
```diff
@@ -54,6 +54,7 @@ export interface ILiveChatWidgetProps {
     preChatSurveyPaneProps?: IPreChatSurveyPaneProps;
     proactiveChatPaneProps?: IProactiveChatPaneStatefulProps;
     reconnectChatPaneProps?: IReconnectChatPaneStatefulProps;
+    errorPaneProps?: ILoadingPaneProps;
     startChatErrorPaneProps?: IStartChatErrorPaneProps;
```
Comment (charliewang95) [inline chat-widget/src/components/livechatwidget/interfaces/ILiveChatWidgetProps.ts:58]: Why do we have `startChatErrorPaneProps` as well as `errorPaneProps`? Do we have other error pane types?

Comment (charliewang95) [reply chat-widget/src/components/livechatwidget/interfaces/ILiveChatWidgetProps.ts:58]: From the props shown in description, doesn't it make more sense to put the controlProps of `startChatErrorPaneProps` inside `errorPaneProps`?

File: chat-widget/src/components/startchaterrorpanestateful/StartChatErrorPaneStateful.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    const [state, ]: [ILiveChatWidgetContext, Dispatch<ILiveChatWidgetAction>] = useChatContextStore();
+    const { errorPaneProps, startChatErrorPaneProps } = props;
+    
+    const generalStyleProps: IStyle = Object.assign({}, defaultErrorPaneGeneralStyleProps, errorPaneProps?.styleProps?.generalStyleProps);
+    const titleStyleProps: IStyle = Object.assign({}, defaultErrorPaneTitleStyleProps, errorPaneProps?.styleProps?.titleStyleProps);
+    const subtitleStyleProps: IStyle = Object.assign({}, defaultErrorPaneSubtitleStyleProps, errorPaneProps?.styleProps?.subtitleStyleProps);
+    const iconStyleProps: IStyle = Object.assign({}, defaultErrorPaneIconStyleProps, errorPaneProps?.styleProps?.iconStyleProps);
+    const iconImageProps: IImageProps = Object.assign({}, defaultErrorPaneIconImageStyleProps, errorPaneProps?.styleProps?.iconImageProps);
+
+    const errorUIStyleProps: ILoadingPaneStyleProps = {
```
Comment (charliewang95) [inline chat-widget/src/components/startchaterrorpanestateful/StartChatErrorPaneStateful.tsx:30]: Seems like we are purely using `errorPaneProps` for styles and `startChatErrorPaneProps` for controls (strings). Is that by design?

The reason behind this is: will it cause confusion - if I add props for `errorPaneProps.controlProps`, it actually doesn't do anything right?

File: chat-widget/src/contexts/common/LiveChatWidgetContextInitialState.ts [hunk 3]
```diff
@@ -36,7 +36,7 @@ export const getLiveChatWidgetContextInitialState = (props: ILiveChatWidgetProps
             initialChatSdkRequestId: "",
             transcriptRequestId: "",
             confirmationPaneConfirmedOptionClicked: false,
-            confirmationState: ConfirmationState.NotSet
+            confirmationState: ConfirmationState.NotSet,
```
Comment (charliewang95) [inline chat-widget/src/contexts/common/LiveChatWidgetContextInitialState.ts:39]: nit; remove comma

File: chat-widget/src/components/startchaterrorpanestateful/StartChatErrorPaneStateful.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        generalStyleProps: generalStyleProps,
+        titleStyleProps: titleStyleProps,
+        subtitleStyleProps: subtitleStyleProps,
+        iconStyleProps: iconStyleProps,
+        iconImageProps: iconImageProps,
+    };
+
+    const errorPaneTitleText = startChatErrorPaneProps?.controlProps?.titleText ?? "We are unable to load chat at this time.";
```
Comment (xTEddie) [inline chat-widget/src/components/startchaterrorpanestateful/StartChatErrorPaneStateful.tsx:38]: Can the strings be extracted to some kind of default values as `const`?

File: chat-widget/src/components/startchaterrorpanestateful/StartChatErrorPaneStateful.tsx (hunk 2 above)
Comment (kevinlin98z) [reply chat-widget/src/components/startchaterrorpanestateful/StartChatErrorPaneStateful.tsx:29]: Did some refactoring -- `errorPaneProps` is now consolidated into `startChatErrorPaneProps`, with `startChatErrorPaneProps.controlProps` controlling all the associated props with this new pane. 

Additional `StartChatErrorPane` error string scenarios will be added within `startChatErrorPaneProps.controlProps` such as `authTitleText`, `authSubtitleText`, `[scenario]TitleText`, `[scenario]SubtitleText`.

File: chat-widget/src/components/livechatwidget/interfaces/ILiveChatWidgetProps.ts (hunk 1 above)
Comment (kevinlin98z) [reply chat-widget/src/components/livechatwidget/interfaces/ILiveChatWidgetProps.ts]: Refactored, see other comment.

[the remark this answers is not in the document]
Comment (kevinlin98z) [reply chat-widget/src/components/livechatwidget/interfaces/ILiveChatWidgetProps.ts]: Refactored, see other comment.

File: chat-widget/src/components/startchaterrorpanestateful/StartChatErrorPaneStateful.tsx (hunk 4 above)
Comment (kevinlin98z) [reply chat-widget/src/components/startchaterrorpanestateful/StartChatErrorPaneStateful.tsx]: Moved to constants file.

File: chat-widget/src/common/Constants.ts [hunk 5]
```diff
@@ -304,6 +304,12 @@ export class NotificationPaneConstants {
     public static readonly ChromeCloseIconName = "ChromeClose";
 }
 
+export class StartChatErrorPaneConstants {
+    public static readonly DefaultStartChatErrorPaneId = "oc-lcw-start-chat-error-pane";
+    public static readonly DefaultStartChatErrorTitleText = "We are unable to load chat at this time.";
```
Comment (elopezanaya) [inline chat-widget/src/common/Constants.ts:309]: we need to localize default messages?

Comment (kevinlin98z) [reply chat-widget/src/common/Constants.ts:309]: Localization will happen in the LCW OOB layer, the changes will be made there.

File: chat-widget/src/components/startchaterrorpanestateful/StartChatErrorPaneStateful.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        titleStyleProps: titleStyleProps,
+        subtitleStyleProps: subtitleStyleProps,
+        iconStyleProps: iconStyleProps,
+        iconImageProps: iconImageProps,
+    };
+
+    const errorPaneTitleText = startChatErrorPaneProps?.controlProps?.titleText ?? StartChatErrorPaneConstants.DefaultStartChatErrorTitleText;
+    const errorPaneSubtitleText = startChatErrorPaneProps?.controlProps?.subtitleText ?? StartChatErrorPaneConstants.DefaultStartChatErrorSubtitleText;
+
+    const errorUIControlProps: ILoadingPaneControlProps = {
```
Comment (elopezanaya) [inline chat-widget/src/components/startchaterrorpanestateful/StartChatErrorPaneStateful.tsx:41]: can we confirm if screen readers are describing elements in the pane properly?, or if we need to add aria-labels

File: chat-widget/src/components/startchaterrorpanestateful/common/defaultErrorPaneIconImageProps.ts [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import { IImageProps } from "@fluentui/react";
+import { ImageFit } from "@fluentui/react";
+import { AlertIcon } from "../../../assets/Icons";
+
+export const defaultErrorPaneIconImageStyleProps: IImageProps = {
```
Comment (elopezanaya) [inline chat-widget/src/components/startchaterrorpanestateful/common/defaultErrorPaneIconImageProps.ts:5]: could you confirm look&feel is uniform in Android and iOS devices? using chrome, safari, edge

**Review by elopezanaya — COMMENTED** [review]
pending corroboration to be added as test evidence

File: chat-widget/src/components/startchaterrorpanestateful/StartChatErrorPaneStateful.tsx (hunk 6 above)
Comment (kevinlin98z) [reply chat-widget/src/components/startchaterrorpanestateful/StartChatErrorPaneStateful.tsx:41]: Screen reader will read the text on the pane.

File: chat-widget/src/components/startchaterrorpanestateful/common/defaultErrorPaneIconImageProps.ts (hunk 7 above)
Comment (kevinlin98z) [reply chat-widget/src/components/startchaterrorpanestateful/common/defaultErrorPaneIconImageProps.ts:5]: Tested on few android / iOS devices, see below for some samples: https://microsoft-my.sharepoint.com/:f:/p/kevinlin/Eq6xXZf4pElGqWnyzt783BEBFwnRFOohp881DwptHnf9Lw?e=tcW8xG

File: chat-widget/src/components/startchaterrorpanestateful/interfaces/IStartChatErrorPaneClassNames.ts [hunk 8]
```diff
@@ -0,0 +1,5 @@
+export interface IStartChatErrorPaneClassNames {
```
Comment (charliewang95) [inline chat-widget/src/components/startchaterrorpanestateful/interfaces/IStartChatErrorPaneClassNames.ts:1]: Please update dev guide on the new interfaces

Comment (charliewang95) [inline chat-widget/src/components/startchaterrorpanestateful/interfaces/IStartChatErrorPaneClassNames.ts:1]: Please update change log

File: CHANGE_LOG.md [hunk 9]
```diff
@@ -6,6 +6,9 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
+### Added
+- Enabled customizations for the start chat error pane
```
[the remark this answers is not in the document]
Comment (kevinlin98z) [reply CHANGE_LOG.md]: Added dev docs

File: chat-widget/src/components/startchaterrorpanestateful/interfaces/IStartChatErrorPaneClassNames.ts (hunk 8 above)
Comment (kevinlin98z) [reply chat-widget/src/components/startchaterrorpanestateful/interfaces/IStartChatErrorPaneClassNames.ts:1]: Added dev docs

Comment (kevinlin98z) [reply chat-widget/src/components/startchaterrorpanestateful/interfaces/IStartChatErrorPaneClassNames.ts:1]: Updated to reflect `IStartChatErrorPaneProps` being added.

**Review by xTEddie — APPROVED** [review]

**Review by charliewang95 — APPROVED** [review]

**Pull request merged**